lib/em-jack/connection.rb

```diff
@@ -322,13 +324,10 @@ def disconnected
     end
 
     def reconnect!
-      @retries = 0
-
-      reset_tube_state
-      EM.next_tick do
-        @conn.reconnect(@host, @port)
-        initialize_tube_state
-      end
```

Comment:
Do you know what the lines you removed do? Have you checked whether we use this method for anything?

Reply:
The previous `reconnect!` method:

- reset the retries count (this happens in the `connected` handler, which I think is right)
- resets the tube state (this happens in the `disconnected` handler, which will be called when the disconnection which we trigger occurs)
- Calls `reconnect` on the `@conn` which will do nothing if the state is still `connected`
- initialises the tube state (this also happens in the `disconnected` handler

I don't think any of the stuff that happens in the old method won't happen with the new method!

As for whether anything uses it - I can't see anything from `grep`-ing in pusher-server and jobsworth!

It's also been running on `eu` overnight and everything seems fine!